src/Integration.Vsix/CFamily/VcxProject/VcxRequestFactory.cs

```diff
@@ -0,0 +1,161 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.CFamily.Analysis;
+using SonarLint.VisualStudio.Core.CFamily;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject
+{
+    [Export(typeof(IRequestFactory))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class VcxRequestFactory : IRequestFactory
+    {
+        private readonly ICFamilyRulesConfigProvider cFamilyRulesConfigProvider;
+        private readonly IRulesConfigProtocolFormatter rulesConfigProtocolFormatter;
+        private readonly IFileConfigProvider fileConfigProvider;
+        private readonly ILogger logger;
+        private readonly DTE dte;
+
+        [ImportingConstructor]
+        public VcxRequestFactory([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            ICFamilyRulesConfigProvider cFamilyRulesConfigProvider,
+            ILogger logger)
+            : this(serviceProvider,
+                cFamilyRulesConfigProvider,
+                new RulesConfigProtocolFormatter(),
+                new FileConfigProvider(logger),
+                logger)
+        {
+        }
+
+        internal VcxRequestFactory(IServiceProvider serviceProvider,
+            ICFamilyRulesConfigProvider rulesConfigProvider,
+            IRulesConfigProtocolFormatter rulesConfigProtocolFormatter,
+            IFileConfigProvider fileConfigProvider,
+            ILogger logger)
+        {
+            this.dte = serviceProvider.GetService<DTE>();
+            this.cFamilyRulesConfigProvider = rulesConfigProvider;
+            this.rulesConfigProtocolFormatter = rulesConfigProtocolFormatter;
+            this.fileConfigProvider = fileConfigProvider;
+            this.logger = logger;
+        }
+
+        public IRequest TryGet(string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions)
+        {
+            try
+            {
+                var projectItem = dte?.Solution?.FindProjectItem(analyzedFilePath);
```

Comment:
I copied the code as-is from CFamilyHelper but with a few changes:
1. Using FileConfigProvider and RulesConfigProtocolFormatter
2. Wrapping it with a try-catch, so the top-level aggregate factory wouldn't fail when iterating over the factories
3. I removed the ` if (request?.File == null)` condition -- it's not possible for this to happen without an exception and it wasn't tested in the original CFamilyHelperTests.
4. We no longer save the file config here, as it's now part of the caller CLangAnalyzer
5. Technically the try-catch will use the real logger, not the fake NoOp -- but this is ok, because there was no try-catch in the CFamilyHelper to begin with, so any exceptions would've propagated to CLangAnalyzer and written with a real logger.

Reply:
(4) was done already, wasn't it? Has anything else changed about saving the config in this PR?


```
  5. Technically the try-catch will use the real logger, not the fake NoOp -- but this is ok, because there was no try-catch in the CFamilyHelper to begin with, so any exceptions would've propagated to CLangAnalyzer and written with a real logger.
```

Both `CFamilyHelper.IsFileInSolution` and `CFamilyHelper.TryGetConfig` had exception handling before which would have used the no-op logger. 

I thought that the no-op logger was added because the logged output was very verbose for unsupported project types, and lots of messages got logged for PCH generation (although with the addition of CMake support this should happen less often).

I guess that is the real question - what now gets logged for PCH file generation for unsupported files?

Reply:
IsFileInSolution and TryGetConfig are now both part of FileConfigProvider, which has the NoOp logger. So that behavior remains the same -- no verbose output for PCH.

Reply:
(4) no, this was just to clarify that the code existed in CFamilyHelper and was not ported.

Reply:
Ok. I found the comment confusing because it's about something that was done previously, not something that changed in this PR.